yt_dlp/extractor/redgifs.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..compat import compat_parse_qs
+from ..utils import ExtractorError, int_or_none
+
+
+def _parse_gif_entry(gif_data):
+    video_id = gif_data["id"]
+    title = " ".join(gif_data.get('tags') or [])
+    formats = [
+        {
+            "format_id": "gif",
+            "url": gif_data["urls"]["gif"]
+        },
+        {
+            "format_id": "sd",
+            "url": gif_data["urls"]["sd"]
+        },
+        {
+            "format_id": "hd",
+            "url": gif_data["urls"]["hd"]
+        }
+    ]
+    webpage_url = "https://redgifs.com/watch/{}".format(video_id)
+
+    return {
+        'id': video_id,
+        'title': title,
+        'age_limit': 18,
+        'formats': formats,
+        'timestamp': gif_data.get('createDate'),
+        'view_count': int_or_none(gif_data.get('views')),
+        'uploader': gif_data.get('userName'),
+        'webpage_url': webpage_url,
+        'tags': gif_data.get('tags')
+    }
```

Comment:
imo should still be a class function

Reply:
nvmd, i'll wait for https://github.com/yt-dlp/yt-dlp/pull/1631

Reply:
It's a class function now, in RedGifsIE, which the other extractors inherit from